Source/ASNetworkImageNode.mm

```diff
@@ -185,14 +198,18 @@ - (void)setDefaultImage:(UIImage *)defaultImage
   _defaultImage = defaultImage;
 
   if (!_imageLoaded) {
-    BOOL hasURL = _URL == nil;
-    /* We want to maintain the order that currentImageQuality is set regardless of the calling thread,
-     so always use a dispatch_async to ensure that we queue the operations in the correct order.
-     (see comment in displayDidFinish) */
-    dispatch_async(dispatch_get_main_queue(), ^{
-      self.currentImageQuality = hasURL ? 0.0 : 1.0;
-    });
-    [self _setImage:defaultImage];
+    BOOL hasURL = (_URL == nil);
+    {
+      ASDN::MutexUnlocker l(__instanceLock__);
+      /* We want to maintain the order that currentImageQuality is set regardless of the calling thread,
+       so always use a dispatch_async to ensure that we queue the operations in the correct order.
+       (see comment in displayDidFinish) */
+      dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Looking into `setURL:resetToDefault:` - the order we set the `currentImageQuality` and call `_setImage` is different. Wondering if this was intentionally @garrettmoon ?